doc/sphinx/language/core/definitions.rst

```diff
@@ -1,5 +1,5 @@
-Definitions
-===========
+Definitions and theorems
```

Comment:
Makes sense, though the subtitle just above the Theorem command is "Assertions and proofs".  Maybe make that title "Theorems and proofs"??  Always good to have clear relationships between main titles and subtitles that minimize the need to think.

Off topic: the Type Cast section doesn't belong in this chapter.

Reply:
"Theorems and proofs": done.